File: recycleradapter/src/main/java/net/gotev/recycleradapter/SwipeListener.java
package net.gotev.recycleradapter;

import android.support.v7.widget.helper.ItemTouchHelper;

/**
 * Listener methods called when the user performs swipe gesture on one of the adapter items.
 * @author Aleksandar Gotev
 */
public interface SwipeListener {
    /**
     * Called when the user swipes out one element
     * @param position position in the adapter
     * @param direction direction. Refer to {@link ItemTouchHelper} constants.
     */
    void onItemSwiped(int position, int direction);
}

File: recycleradapter-extensions/src/main/java/net/gotev/recycleradapter/ext/SwipeListener.kt
package net.gotev.recycleradapter.ext

/**
 * Listener methods called when the user performs swipe gesture on one of the adapter items.
 * @author Aleksandar Gotev
 */
interface SwipeListener {
    /**
     * Called when the user swipes out one element
     * @param position position in the adapter
     * @param direction direction. Refer to ItemTouchHelper class constants.
     */
    fun onItemSwiped(position: Int, direction: Int)
}
